include/picolibrary/wiznet/w5500/network_stack.h

```diff
@@ -45,6 +45,734 @@ namespace picolibrary::WIZnet::W5500 {
 template<typename Driver>
 class Network_Stack {
   public:
+    /**
+     * \brief TCP over IP socket.
+     */
+    class TCP_Socket {
```

Comment:
Graceful disconnection is not currently handled. May require addition of a new error code and revision of the TCP over IP client/server socket concepts.